### PR TITLE
#175636862 Add Functionality For Travel Admin To Apply Crud on Accommodation Facilities

### PR DESCRIPTION
**Pull request by Thiernope**
#### What does this PR do?
Create, update, get and delete accommodations
#### Description of Task to be completed?

- create userRoles models
- add column of roleId in the User model
- create superAdmin role
- Add functionalities to create, read, update and delete users' roles using their emails
- Restrict the deletion of SUPER_ADMIN role in the system

#### How should this be manually tested?

- run `git clone https://github.com/atlp-rwanda/tech-strikers-bn-backend.git`
- run `cd directory tech-strikers-bn-backend`
- run `git checkout ft-crud-accommodations-#175636862`
- run `npm install`
- run `npm run role:create`
- run `npm run admin:create`
- assign a user to the role called Travel Administrator
- login the user who is already assigned the role of Travel Administrator
- pass his token in the headers and manage all the endpoints
#### Any background context you want to provide?
You should make sure you assign a user to the role of Travel Admin (back to the role-setting feature).
#### What are the relevant pivotal tracker stories?
[#175636862](https://www.pivotaltracker.com/story/show/175636862)
#### Screenshots (if appropriate)
Create Accommodation
Screen Shot 2021-01-19 at 10 08 08

Create a Room
Screen Shot 2021-01-19 at 10 09 08

Get Accommodations
Screen Shot 2021-01-19 at 10 12 10

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -40,5 +40,8 @@
 	"Comment submitted": "Comment submitted",
 	"Request not Found": "Request not Found",
 	"Server Error": "Server Error",
-	"Comment deleted": "Comment deleted"
-}
```
Comment (roclena) [inline locales/en.json]: Add new line

File: src/controllers/room.controller.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            message: res.__("Room updated successfully"), updatedRoom
+        })
+            }
+         
+        }
+
+        
+    
+
+export default CrudRooms
```
Comment (roclena) [inline src/controllers/room.controller.js]: add new line

File: src/database/models/room.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    },{}
+  );
+  Rooms.associate = models =>{
+    Rooms.belongsTo(models.Accommodation, {
+      foreignKey: "accommodationId",
+    });
+  }
+
+  return Rooms;
+};
```
Comment (roclena) [inline src/database/models/room.js]: add new line

Comment (roclena) [inline src/database/models/room.js]: add new line

File: src/validation/crud.accommodation.validations.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        
+            if(!price){
+                return res.status(400).send({error: res.__("Price is required")});
+            }
+        
+            return next();
+            }
+ }
+
+export default AccommodationValidation
```
Comment (roclena) [inline src/validation/crud.accommodation.validations.js:50]: Add new ine

**Review by roclena — APPROVED** [review]

**Review by hervebutera — APPROVED** [review]
Add the new lines as suggested by @roclena and always avoid putting in controllers what should be in services. e.g `.findAll()`

**Review by geraudirad — APPROVED** [review]

**Review by bigelio — APPROVED** [review]

**Pull request merged**